packages/platform-android/src/commands/runAndroid/index.ts

```diff
@@ -324,15 +324,19 @@ function startServerInNewWindow(
     }
   }
   if (process.platform === 'linux') {
-    try {
-      return execa.sync(terminal, ['-e', `sh ${launchPackagerScript}`], {
-        ...procConfig,
-        detached: true,
-      });
-    } catch (error) {
-      // By default, the child shell process will be attached to the parent
-      return execa.sync('sh', [launchPackagerScript], procConfig);
-    }
+    // Awaiting this causes the CLI to hang indefinitely, so this must execute without .sync.
+    return execa(terminal, ['-e', 'sh', launchPackagerScript], {
+      ...procConfig,
+      detached: true,
+    }).catch(_ => {
+      // If the terminal could not be opened, fall back to running it inside the process.
+      const processSh = execa('sh', [launchPackagerScript], procConfig);
```

Comment:
we should guard this with try/catch as well

Reply:
Actually no - the function above catches the error already, which will show a nice error for it. It didn't fully work, but since I remade this function to be async/await, I changed it to be a promise-style catch (with a comment to explain while). Push incoming